chat-widget/src/hooks/useSetCustomContext.ts

```diff
@@ -0,0 +1,83 @@
+import { LogLevel, TelemetryEvent } from "../common/telemetry/TelemetryConstants";
+import { getStateFromCache, getWidgetCacheIdfromProps, isNullOrEmptyString, isUndefinedOrEmpty } from "../common/utils";
+
+import { Constants } from "../common/Constants";
+import { Dispatch } from "react";
+import { ILiveChatWidgetAction } from "../contexts/common/ILiveChatWidgetAction";
+import { ILiveChatWidgetContext } from "../contexts/common/ILiveChatWidgetContext";
+import { ILiveChatWidgetProps } from "../components/livechatwidget/interfaces/ILiveChatWidgetProps";
+import { TelemetryHelper } from "../common/telemetry/TelemetryHelper";
+import useChatContextStore from "./useChatContextStore";
+
+const useSetCustomContext = (props: ILiveChatWidgetProps, widgetInstanceId?: string) => {
+    const [state, ]: [ILiveChatWidgetContext, Dispatch<ILiveChatWidgetAction>] = useChatContextStore();
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const getInitContextParamsForPopout = async (): Promise<any> => {
+        return window.opener ? await getInitContextParamForPopoutFromOuterScope(window.opener) : null;
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const getInitContextParamForPopoutFromOuterScope = async (scope: any): Promise<any> =>  {
+        let payload;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let waitPromiseResolve: any;
+        const waitPromise = new Promise((res, rej) => {
+            waitPromiseResolve = res;
+            setTimeout(() => rej("Failed to find method in outer scope"), 5000);
```

Comment:
why not configurable the 5 seconds wait? , maybe some customers want a shorter wait or longer...

Reply:
This (custom context for popout) is a feature we need to revisit as a whole. Is not related to this PR